File: codes/jiheon/739_dailytemperatures.rs
impl Solution {
    pub fn daily_temperatures(temperatures: Vec<i32>) -> Vec<i32> {
        let mut answer = vec![0i32; temperatures.len()];
        let mut stack: Vec<usize> = Vec::new();

        for (today_idx, &today_temperature) in temperatures.iter().enumerate() {
            while let Some(&prev_day_idx) = stack.last() {
                if temperatures[prev_day_idx] >= today_temperature {
                    break;
                }
                answer[prev_day_idx] = (today_idx - prev_day_idx) as _;
                stack.pop();
            }
            stack.push(today_idx);
        }

        answer
    }
}

File: codes/naeun/739_dailytemperatures.java
class Solution {
    public int[] dailyTemperatures(int[] temperatures) {
        int[] answer = new int[temperatures.length];
        
        for(int start = 0; start < temperatures.length; start++){
            for(int end = start+1; end < temperatures.length; end++){
                if(temperatures[start] < temperatures[end]){
                    answer[start] = end-start;
                    break;
                }
            }
        }
        
        return answer;
    }
}